### PR TITLE
Fixing powershell linting warnings

### LINKED ISSUE #2453: [Windows] Numerous Powershell linting issues in provision.ps1
```
C:\Users\Nick\Documents\GitHub\poppy-osquery\tools [master ≡]> Invoke-ScriptAnalyzer .\provision.ps1

RuleName                            Severity     FileName   Line  Message
--------                            --------     --------   ----  -------
PSAvoidUsingInvokeExpression        Warning      provision. 51    Invoke-Expression is used. Please remove Invoke-Expression
                                                 ps1              from script and find other options instead.
PSAvoidUsingWriteHost               Warning      provision. 45    File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 48    File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 50    File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 53    File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 56    File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 68    File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 71    File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 77    File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 82    File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 85    File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 87    File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 92    File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 95    File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 99    File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 103   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 106   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 109   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 112   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 118   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 148   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 151   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 154   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 157   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 161   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 166   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 169   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 178   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 182   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 186   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 190   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 191   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 193   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 196   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSAvoidUsingWriteHost               Warning      provision. 207   File 'provision.ps1' uses Write-Host. Avoid using
                                                 ps1              Write-Host because it might not work in all hosts, does not
                                                                  work when there is no host, and (prior to PS 5.0) cannot be
                                                                  suppressed, captured, or redirected. Instead, use
                                                                  Write-Output, Write-Verbose, or Write-Information.
PSPossibleIncorrectComparisonWithNu Warning      provision. 46    $null should be on the left side of equality comparisons.
ll                                               ps1
PSPossibleIncorrectComparisonWithNu Warning      provision. 94    $null should be on the left side of equality comparisons.
ll                                               ps1
PSPossibleIncorrectComparisonWithNu Warning      provision. 98    $null should be on the left side of equality comparisons.
ll                                               ps1
PSPossibleIncorrectComparisonWithNu Warning      provision. 177   $null should be on the left side of equality comparisons.
ll                                               ps1
PSUseApprovedVerbs                  Warning      provision. 22    The cmdlet 'Check-ChocoPackage-Installed' uses an
                                                 ps1              unapproved verb.
PSAvoidUsingCmdletAliases           Warning      provision. 51    'iex' is an alias of 'Invoke-Expression'. Alias can
                                                 ps1              introduce possible problems and make scripts hard to
                                                                  maintain. Please consider changing alias to its full
                                                                  content.
PSAvoidUsingCmdletAliases           Warning      provision. 183   'pushd' is an alias of 'Push-Location'. Alias can introduce
                                                 ps1              possible problems and make scripts hard to maintain. Please
                                                                  consider changing alias to its full content.
PSAvoidUsingCmdletAliases           Warning      provision. 185   'popd' is an alias of 'Pop-Location'. Alias can introduce
                                                 ps1              possible problems and make scripts hard to maintain. Please
                                                                  consider changing alias to its full content.
PSUseShouldProcessForStateChangingF Warning      provision. 176   Function ’Update-GitSubmodule’ has verb that could change
unctions                                         ps1              system state. Therefore, the function has to support
                                                                  'ShouldProcess'.
PSUseSingularNouns                  Warning      provision. 91    The cmdlet 'Install-PipPackages' uses a plural noun. A
                                                 ps1              singular noun should be used instead.
PSUseSingularNouns                  Warning      provision. 117   The cmdlet 'Install-ThirdPartyPackages' uses a plural noun.
                                                 ps1              A singular noun should be used instead.

```

### PR DESCRIPTION
**Pull request by PoppySeedPlehzr**
This closes #2453

### CODE REVIEW

File: tools/provision.ps1 [hunk 1]
```diff
@@ -174,19 +194,29 @@ function Install-ThirdPartyPackages {
 }
 
 function Update-GitSubmodule {
-  if ((Get-Command 'git.exe' -ErrorAction SilentlyContinue) -eq $null) {
+  [CmdletBinding(SupportsShouldProcess=$true,ConfirmImpact="Low")]
+  param()
+  if(-not $PSCmdlet.ShouldProcess('Git Submodules')){
```
Comment (theopolis) [inline tools/provision.ps1]: You have spaces in the other conditionals before `-not`.

File: tools/provision.ps1 [hunk 2]
```diff
@@ -174,19 +194,29 @@ function Install-ThirdPartyPackages {
...
     Write-Host "[-] ERROR: Git was not found on the system. Install git." -foregroundcolor Red
     Exit -1
   }
   $thirdPartyPath = Resolve-Path ([System.IO.Path]::Combine($PSScriptRoot, '..', 'third-party'))
   Write-Host " => Updating git submodules in $thirdPartyPath ..." -foregroundcolor Yellow
-  pushd $thirdPartyPath
+  Push-Location $thirdPartyPath
   git submodule update --init
-  popd
+  Push-Location
```
Comment (theopolis) [inline tools/provision.ps1]: This is going from a `popd` to a `Push`?

File: tools/provision.ps1 [hunk 3]
```diff
@@ -174,19 +194,29 @@ function Install-ThirdPartyPackages {
...
   Write-Host "[+] Submodules updated!" -foregroundcolor Yellow
 }
 
 function Main {
+  if( $PSVersionTable.PSVersion.Major -lt 3.0 ){
```
Comment (theopolis) [inline tools/provision.ps1]: nit, spaces here, before and after. In other places you have `if(THING)`.

**Comment by facebook-github-bot-2** [comment]
@PoppySeedPlehzr updated the pull request - [view changes](https://github.com/facebook/osquery/pull/2470/files/864f9dc73a331b601a948b975afe7c697c3b4371..decff81c3549b95ec86791559b719c2ff8496cdc)

**Comment by osqueryer** [comment]
:-1: The commit decff81c3549b95ec86791559b719c2ff8496cdc (Job results: [3276](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3276/)) failed one or more tests (**OS X/Linux**).

**Pull request merged**